tools/Dockerfile.reproducible

```diff
@@ -0,0 +1,105 @@
+    ninja install
+
+RUN tar xf llvm-project-${_LEAP_CLANG_VERSION}.src.tar.xz && \
+    cmake -S llvm-project-${_LEAP_CLANG_VERSION}.src/llvm -B build-toolchain -GNinja -DLLVM_INCLUDE_DOCS=Off -DLLVM_TARGETS_TO_BUILD=host -DCMAKE_BUILD_TYPE=Release \
+                                                                                     -DCMAKE_INSTALL_PREFIX=/pinnedtoolchain \
+                                                                                     -DCOMPILER_RT_BUILD_SANITIZERS=Off \
+                                                                                     -DLLVM_ENABLE_PROJECTS='lld;clang;clang-tools-extra' \
+                                                                                     -DLLVM_ENABLE_RUNTIMES='compiler-rt;libc;libcxx;libcxxabi;libunwind' && \
+    cmake --build build-toolchain -t install
+
+COPY <<-"EOF" /pinnedtoolchain/pinnedtoolchain.cmake
+   set(CMAKE_C_COMPILER ${CMAKE_CURRENT_LIST_DIR}/bin/clang)
+   set(CMAKE_CXX_COMPILER ${CMAKE_CURRENT_LIST_DIR}/bin/clang++)
+
+   set(CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_LIST_DIR}/include/c++/v1 ${CMAKE_CURRENT_LIST_DIR}/include/x86_64-unknown-linux-gnu/c++/v1 /usr/local/include /usr/include)
+
+   set(CMAKE_C_FLAGS_INIT "-D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie -pthread")
+   set(CMAKE_CXX_FLAGS_INIT "-nostdinc++ -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie -pthread")
+
+   set(CMAKE_EXE_LINKER_FLAGS_INIT "-stdlib=libc++ -nostdlib++ -pie -pthread -Wl,-z,relro,-z,now")
+   set(CMAKE_SHARED_LINKER_FLAGS_INIT "-stdlib=libc++ -nostdlib++")
+   set(CMAKE_MODULE_LINKER_FLAGS_INIT "-stdlib=libc++ -nostdlib++")
+
+   set(CMAKE_CXX_STANDARD_LIBRARIES "${CMAKE_CURRENT_LIST_DIR}/lib/x86_64-unknown-linux-gnu/libc++.a ${CMAKE_CURRENT_LIST_DIR}/lib/x86_64-unknown-linux-gnu/libc++abi.a")
+
+   set(CMAKE_SYSTEM_PREFIX_PATH "${CMAKE_CURRENT_LIST_DIR}/pinllvm")
+EOF
+ENV CMAKE_TOOLCHAIN_FILE=/pinnedtoolchain/pinnedtoolchain.cmake
+
+RUN tar xf llvm-project-${_LEAP_LLVM_VERSION}.src.tar.xz && \
+    cmake -S llvm-project-${_LEAP_LLVM_VERSION}.src/llvm -B build-pinllvm -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=host -DLLVM_BUILD_TOOLS=Off \
+                                                                                  -DLLVM_ENABLE_RTTI=On -DLLVM_ENABLE_TERMINFO=Off -DLLVM_ENABLE_PIC=Off \
+                                                                                  -DCMAKE_INSTALL_PREFIX=/pinnedtoolchain/pinllvm && \
+    cmake --build build-pinllvm -t install
+
+RUN rm -rf llvm* build* cmake*
+
+FROM builder AS build
+
+ARG LEAP_BUILD_JOBS
+
+COPY / /src
+RUN cmake -S src -B build -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -GNinja && \
+    cmake --build build -t package -- ${LEAP_BUILD_JOBS:+-j$LEAP_BUILD_JOBS} && \
+    src/tools/tweak-deb.sh build/leap_*.deb
+
+FROM scratch AS exporter
+COPY --from=build /build/*.deb /build/*.tar.* /
```

Comment:
A deficiency here is that users cannot (easily) run unit tests themselves. Possibly a wiki page or other such documentation should be added with the additional complex steps to run the tests. (you would need to build an image with the `build` target, and then run a container with it, and then run `ctest` inside it)